Refactor/#121 재료 저장, 수정 코드 리펙토링

juryheed opened the pull request:
## Description
매개 변수 수정
저장방법 결정 메서드 수행 위치 변경

## Changes
- [x] FridgeService
- [x] FridgeItem엔티티

## Additional context
- [x] 존재하지 않는 저장방법을 입력시 발생하는 ErrorCode

closed #121

jher235 committed:
Merge pull request #106 from mju-likelion/develop

변경사항 적용 후 재배포

jher235 committed:
Merge pull request #113 from mju-likelion/develop

변경사항 적용 후 재배포

juryheed committed:
Merge branch 'main' of https://github.com/mju-likelion/12th-hackathon-team1-server into develop

juryheed committed:
refactor: 변수 변경- #121
- 메서드에서 Dto대신 직접 Id를 받도록 수정

juryheed committed:
refactor: 저장방법 결정 메서드를 엔티티에서 처리하도록 변경 - #121
- Entity로 옮기면서 메서드명 변경
- 상온, 냉동, 냉장 외 다른 Strign이 들어오면 예외가 발생함

jher235 commented on src/main/java/com/hackathonteam1/refreshrator/service/FridgeService.java:
```diff
@@ -140,19 +140,9 @@ public FridgeItemResponseData detailIngredientInFridge(UUID fridgeItemId, User u
     //데이터베이스에서 재료id로 재료를 찾는 메서드
-    private Ingredient findIngredient(AddFridgeDto addFridgeDto){
-        return ingredientRepository.findById(addFridgeDto.getIngredientId())
+    private Ingredient findIngredient(UUID ingredientId){
```
개인적인 생각으론 메서드 명이 좀 더 명확하다면 좋을 것 같아요.
근데 대현님이 말씀하신 것 중에 Ingredient를 찾는 메서드는 IngredientService에 작성하고 이를 의존해서 사용하라고 말씀하셨던게 있는데 이 방법은 어떨까요?

juryheed replied:
수정했습니다!

jher235 commented on src/main/java/com/hackathonteam1/refreshrator/exception/errorcode/ErrorCode.java:
```diff
@@ -39,6 +39,7 @@ public enum ErrorCode {
     PAGE_NOT_FOUND("4046", "페이지를 찾을 수 없습니다"),
     IMAGE_NOT_FOUND("4047","이미지를 찾을 수 없습니다"),
     RECIPE_LIKE_NOT_FOUND("4048", "좋아요를 누른 레시피가 아닙니다."),
+    STORAGE_NOT_FOUND("4049","존재하지 않는 저장 방법입니다."),
```
이건 BadRequest가 될 여지도 있는 것 같은데, 혹시 어떻게 생각하세요?

juryheed replied:
저도 그생각을 하긴 했었는데 .... BadRequest가 더 나을거 같네요!!
수정했습니당

jihyeonEom commented on src/main/java/com/hackathonteam1/refreshrator/entity/FridgeItem.java:
```diff
@@ -37,6 +39,17 @@ public class FridgeItem extends BaseEntity{
+            case "상온"-> FridgeItem.Storage.STORE_AT_ROOM_TEMPERATURE;
+            case "냉동"-> FridgeItem.Storage.FROZEN;
+            case "냉장" -> FridgeItem.Storage.REFRIGERATED;
+            default -> throw new NotFoundException(ErrorCode.STORAGE_NOT_FOUND);
```
저희 기능에서 사용자가 냉장고에 재료를 등록할 때, 존재하지 않는 저장 방법을 사용할 수 있는 경우가 있나용??

juryheed replied:
우회적인 방법으로 등록을 하는 경우,  "상온","냉장","냉동" 외의 다른 문장이 들어가도 기존 방법에서는 REFRIGERATED가 default로 등록될수 있어서 고쳐봤어요!

juryheed committed:
refactor: 에러코드 수정 - #121
- 저장방법이 존재하지 않는경우 404에러에서 400에러가 발생하도록 변경함

juryheed committed:
refactor: 에러코드 메시지 문장 변경 - #121

juryheed committed:
refactor: findIngredientById 리펙토링 - #121
- 메서드명을 구체적으로 변경
- 데이터베이스에서 재료id로 재료를 찾는 메서드이기 때문에 IngredientService에 작성을 한 후 이를 의존하여 FridgeService의 addIngredientInFridge에서 사용함

jher235 commented:
머지해도 될 것 같습니다!
